doc/detection.dox

```diff
@@ -0,0 +1,16 @@
+// -*- lsst-c++ -*-
+
+namespace lsst { namespace afw { namespace detection {
+/**
+\section afwDetection Detection Overview
+
+Detected pixels are manipulated using the following classes:
+ - Footprint : A set of pixels above (or below) a Threshold
+ - FootprintFunctor : A functor to process the pixels in a Footprint
+ - FootprintSet : A set of Footprints associated with a MaskedImage
+ - Peak : A peak pixel in the %image
+ - Source : The properties of a source of electrons (e.g. a star or cosmic ray)
+ - Threshold : An object describing the threshold used to define a set of Footprint%s
```

Comment:
This is still Doxygen, not Sphinx. You could create a file similar to https://github.com/lsst/afw/blob/master/doc/lsst.afw.cameraGeom/cameraGeom.rst... though now that I read the text, I think this is all redundant with the listing that Sphinx produces automatically (class descriptions don't show up yet for C++ types, but that's being worked on). So I, at least, would be fine with deleting it entirely.

Reply:
Right, this confused me somewhat because I was taking lsst.afw.table as my model, and it still has a .dox file that looks more or less like the one in question (though it also has an .rst file). I'm also fine with deleting this; I can just add any bits from this file that aren't already in the existing .rst file to the latter.